tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,352 @@ use strict;
     }
+
     wait_screen_change { send_key 'alt-w' };
     for (1 .. 12) { send_key 'backspace'; }
     # give a new name for Workgroup
-    type_string 'QA-Workgroup';
+    type_string($samba_directives->{workgroup});
     assert_screen 'yast2_samba-server_workgroup_new';
     send_key 'alt-n';
 
-    # select "Not a Domain Controller"
-    assert_screen 'yast2_samba_server_selection';
-    send_key 'alt-c';
+    ## Handle Domain Controller in sle12
+    if ((is_sle("<15")) || (is_leap("<15.0"))) {
+        # select "Not a Domain Controller"
+        assert_screen 'yast2_samba_server_selection';
+        send_key 'alt-c';
 
-    # check "Not a DC" is select
-    assert_screen 'yast2_samba-server_not-a-dc_selected';
-    send_key 'alt-n';
-    wait_still_screen;
-    # now move to Samba configuration and check enable service start during boot
-    send_key 'alt-r';
-    assert_screen 'yast2_samber-server_start-during-boot';
+        # check "Not a DC" is select
+        assert_screen 'yast2_samba-server_not-a-dc_selected';
+        send_key 'alt-n';
+    }
 
-    # open port in firewall if it is enabled and check network interfaces, check long text by send key right.
-    assert_screen 'yast2_samba_open_port_firewall';
+    # service starts during boot, wait to load default data
+    assert_screen 'yast2_samba-startup-configuration';
+    wait_screen_change { send_key 'alt-r' };
+    assert_screen 'yast2_samba-server_start-during-boot';
+    # open firewalld port
     wait_screen_change { send_key 'alt-f' };
-    send_key 'alt-i';
-
-    assert_screen 'yast2_samba_firewall_port_details';
-    wait_screen_change { send_key 'alt-e' };
-    wait_screen_change { send_key 'alt-a' };
-    wait_screen_change { send_key 'alt-o' };
+    assert_screen 'yast2_samba_open_port_firewall';
 
-    # switch to Samba Configuration - Shares
-    send_key 'alt-s';
+    ## switch to Samba Configuration - Shares
+    wait_screen_change { send_key 'alt-s' };
+    wait_still_screen(2);
     assert_screen 'yast2_samba-server_shares';
-
     # add a shares config html_public
     wait_screen_change { send_key 'alt-a' };
     assert_screen 'yast2_samba-server_new-share';
-
-    type_string 'html_public';
+    # enter share details
+    type_string('html_public');
     wait_screen_change { send_key 'alt-a' };    # set share description
-    type_string 'html docs for share';
+    type_string($samba_directives->{comment});
     wait_screen_change { send_key 'alt-s' };
-    type_string '/html_public';                 # set share path to /home/html_public
+    for (1 .. 5) { send_key 'backspace'; }
+    type_string($samba_directives->{path});     # set share path to /home/html_public
     wait_screen_change { send_key 'alt-r' };    # set read-only
-
-    # check config before confirm new share with ok, confirm to create new share path
+                                                # check config before confirm new share with ok, confirm to create new share path
```

Comment:
that looks a bit weird, please bring back the blank line to separate. This way the tidy script will keep the comment aligned - I think